lib/ocpp/common/websocket/websocket_tls_tpm.cpp

```diff
@@ -398,24 +349,65 @@ void WebsocketTlsTPM::client_loop() {
 
     info.fd_limit_per_thread = 1 + 1 + 1;
 
-    bool use_tpm = connection_options.use_tpm_tls;
+    // Setup context - need to know the key type first
+
+    std::string path_key;
+    std::string path_chain;
+    std::optional<std::string> password;
 
-    // Setup context
-    OSSL_LIB_CTX* lib_ctx;
-    SSL_CTX* ssl_ctx;
+    if (this->connection_options.security_profile == 3) {
 
-    create_sec_context(use_tpm, lib_ctx, ssl_ctx);
+        const auto certificate_key_pair =
+            this->evse_security->get_key_pair(CertificateSigningUseEnum::ChargingStationCertificate);
 
-    // Connection acquire the contexts
-    conn_data->sec_lib_context = std::unique_ptr<OSSL_LIB_CTX>(lib_ctx);
-    conn_data->sec_context = std::unique_ptr<SSL_CTX>(ssl_ctx);
+        if (!certificate_key_pair.has_value()) {
+            EVLOG_AND_THROW(std::runtime_error(
+                "Connecting with security profile 3 but no client side certificate is present or valid"));
+        }
+
+        path_chain = std::string(certificate_key_pair.value().certificate_path.c_str());
```

Comment:
There is a possible error here. The get_key_pair can return 2 values for a valid certificate, one is certificate_path and another is 'certificate_single_path'. if the certificate_path there should be a fallback to the 'certificate_single_path'.

Reply:
I'll make the change.
However get_key_pair() could be updated so that this isn't required.
What if both certificate and certificate_single have values?
Is there a use case where certificate_single is needed on its own?

Reply:
Yes, there is a case with symlinks in the library. Because most of the times both will be present (chain & single) it is better to have this as both, and one can be empty. If both are non-null then the usage depends on the certificate setup.